hudi-common/src/main/java/org/apache/hudi/internal/schema/convert/AvroInternalSchemaConverter.java

```diff
@@ -154,6 +156,29 @@ public static Type buildTypeFromAvroSchema(Schema schema) {
     return visitAvroSchemaToBuildType(schema, visited, true, nextId);
   }
 
+  private static void checkNullType(Type fieldType, String fieldName, Deque<String> visited) {
```

Comment:
is it tested?

Reply:
Yes, TestAvroSchemaEvolutionUtils.java I added the test testNullFieldType()